Mathlib/Algebra/Group/Irreducible/Defs.lean

```diff
@@ -62,11 +62,16 @@ lemma not_irreducible_one : ¬Irreducible (1 : M) := by simp [irreducible_iff]
 lemma Irreducible.ne_one (hp : Irreducible p) : p ≠ 1 := by rintro rfl; exact not_irreducible_one hp
 
 @[to_additive]
-lemma of_irreducible_mul : Irreducible (x * y) → IsUnit x ∨ IsUnit y | ⟨_, h⟩ => h rfl
+lemma of_irreducible_mul : Irreducible (a * b) → IsUnit a ∨ IsUnit b | ⟨_, h⟩ => h rfl
 
 @[to_additive]
 lemma irreducible_or_factor (hp : ¬IsUnit p) :
     Irreducible p ∨ ∃ a b, ¬IsUnit a ∧ ¬IsUnit b ∧ p = a * b := by
   simpa [irreducible_iff, hp, and_rotate] using em (∀ a b, p = a * b → IsUnit a ∨ IsUnit b)
 
+@[to_additive]
+lemma Irreducible.eq_one_or_eq_one [Subsingleton Mˣ] (hp : Irreducible p) (hab : p = a * b) :
```

Comment:
I'd usually expect the `p` to be substituted out. Any reason not to?

(I'm also reminded of `Nat.not_prime_mul {a b : ℕ} (a1 : a ≠ 1) (b1 : b ≠ 1) : ¬Prime (a * b)`)

Reply:
My case is [here](https://github.com/YaelDillies/Toric/blob/df238dd9c0854949a0303e33170815204aa74fcc/Toric/AffineMonoid/Irreducible.lean#L28) and the proof I provide for `hab` is `rfl`. Still, I believe that it might not be practical to substitute `p` out in all cases, eg because we want to provide the proof on the fly or because `p` isn't a free variable.

Reply:
Actually, this matches `Irreducible.isUnit_or_isUnit`

Reply:
I agree with Ruben, please substitute out `p` here.

Reply:
Probably will get changed back eventually, but okay